app/controlplane/internal/usercontext/attjwtmiddleware/attmiddleware.go

```diff
@@ -0,0 +1,276 @@
+// limitations under the License.
+
+package attjwtmiddleware
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/chainloop-dev/chainloop/app/controlplane/internal/jwt/apitoken"
+	"github.com/chainloop-dev/chainloop/app/controlplane/internal/jwt/robotaccount"
+	"github.com/chainloop-dev/chainloop/app/controlplane/internal/jwt/user"
+	errorsAPI "github.com/go-kratos/kratos/v2/errors"
+	"github.com/go-kratos/kratos/v2/middleware"
+	"github.com/go-kratos/kratos/v2/transport"
+	"github.com/golang-jwt/jwt/v4"
+)
+
+const (
+	// bearerWord the bearer key word for authorization
+	bearerWord string = "Bearer"
+	// authorizationKey holds the key used to store the JWT Token in the request tokenHeader.
+	authorizationKey string = "Authorization"
+	// reason holds the error reason.
+	reason string = "UNAUTHORIZED"
+	// RobotAccountProviderKey is the key for robot account token provider
+	RobotAccountProviderKey = "robotAccountProvider"
+	// APITokenProviderKey is the key for api token provider
+	APITokenProviderKey = "apiTokenProvider"
+)
+
+var (
+	ErrMissingJwtToken        = errorsAPI.Unauthorized(reason, "JWT token is missing")
+	ErrMissingKeyFunc         = errorsAPI.Unauthorized(reason, "keyFunc is missing")
+	ErrTokenInvalid           = errorsAPI.Unauthorized(reason, "Token is invalid")
+	ErrTokenExpired           = errorsAPI.Unauthorized(reason, "JWT token has expired")
+	ErrTokenParseFail         = errorsAPI.Unauthorized(reason, "Fail to parse JWT token ")
+	ErrUnSupportSigningMethod = errorsAPI.Unauthorized(reason, "Wrong signing method")
+	ErrWrongContext           = errorsAPI.Unauthorized(reason, "Wrong context for middleware")
+)
+
+// NewRobotAccountProvider return the configuration to validate and verify token issued for Robot Accounts
+func NewRobotAccountProvider(signingSecret string) JWTOption {
+	return withTokenProvider(
+		RobotAccountProviderKey,
+		WithClaims(func() jwt.Claims { return &robotaccount.CustomClaims{} }),
+		WithVerifyAudienceFunc(func(token *jwt.Token) bool {
+			claims, ok := token.Claims.(*robotaccount.CustomClaims)
+			if !ok {
+				return false
+			}
+			for _, aud := range []string{robotaccount.Audience, robotaccount.DeprecatedAudience} {
+				if claims.VerifyAudience(aud, true) {
+					return true
+				}
+			}
+			return false
+		}),
+		WithSigningMethod(robotaccount.SigningMethod),
+		WithKeyFunc(func(_ *jwt.Token) (interface{}, error) {
+			// TODO: add support to multiple signing methods and keys
+			return []byte(signingSecret), nil
+		}),
+	)
+}
+
+// NewAPITokenProvider return the configuration to validate and verify token issued for API Tokens
+func NewAPITokenProvider(signingSecret string) JWTOption {
+	return withTokenProvider(
+		APITokenProviderKey,
+		WithVerifyAudienceFunc(func(token *jwt.Token) bool {
+			claims, ok := token.Claims.(jwt.MapClaims)
+			if !ok {
+				return false
+			}
+			for _, aud := range []string{apitoken.Audience} {
+				if claims.VerifyAudience(aud, true) {
```

Comment:
this code could be simplified. 

`return claims.VerifyAudience(apitoken.Audience, true)`